subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c

```diff
@@ -662,8 +730,9 @@ static void isr_rx(void *param)
 		lll->bn_curr++;
 
 		/* Check payload buffer overflow */
-		payload_offset = (lll->bn_curr - 1U);
-		if (payload_offset > lll->payload_count_max) {
+		payload_offset = (lll->latency_event * lll->bn) +
+				 (lll->bn_curr - 1U);
```

Comment:
You are adding multiplying 2 uint8_ts and adding it to another uint8 and store the result in a uint8 without any checks for overflow. 

This looks dangerous to me, and suggest to either store the result in a uint16 and check the result, or add an assert for the other values to ensure that we don't get an overflow here.

Reply:
addressing here: https://github.com/zephyrproject-rtos/zephyr/pull/75638